src/main/scala/com/amazon/deequ/profiles/ColumnProfiler.scala

```diff
@@ -347,26 +354,29 @@ object ColumnProfiler {
   private[this] def extractGenericStatistics(
       columns: Seq[String],
       schema: StructType,
-      results: AnalyzerContext)
+      results: AnalyzerContext,
+      predefinedColumnDataTypes: Map[String, DataTypeInstances.Value])
     : GenericColumnStatistics = {
 
     val numRecords = results.metricMap
       .collect { case (_: Size, metric: DoubleMetric) => metric.value.get }
       .head
       .toLong
 
+
     val inferredTypes = results.metricMap
-      .collect { case (analyzer: DataType, metric: HistogramMetric) =>
-        val typeHistogram = metric.value.get
-        analyzer.column -> DataTypeHistogram.determineType(typeHistogram)
+      .collect { case (analyzer: DataType, metric: HistogramMetric)
+        if !predefinedColumnDataTypes.contains(analyzer.column) =>
```

Comment:
please use a filter before the collect

Reply:
Oops! Sorry for not catching this in the first place. Sure will do.